app/helpers/sources_helper.rb

```diff
@@ -13,4 +13,36 @@ def render_media_asset
       render partial: 'video'
     end
   end
+
+  ##
+  # Render a back link to the referring page if the referring page is a guide or
+  # set.  Links from guides or sets to sources may be created by admins, so the
+  # applicaiton cannot expect that back URI's can be included in params.
+  # Therefore, it is necessary to parse the referring URI to see if it is a
+  # guide or set, and render a back link accordingly.
+  #
+  # @return [String] HTML back link
+  def render_back_link
+    return unless request.referer.present?
+    uri = URI(request.referer)
+    # return if referrer is an offsite link
+    return unless uri.host == request.host
+
+    uri_path = uri.path
+    # return if referrer does not have relative url root
+    return unless uri_path.slice!(Settings.relative_url_root).present?
```

Comment:
I was curious about the use of `slice!` because I've always used `slice` functions in other languages / situations for extracting ranges of things, not string comparisons. When I ran something similar through `irb`, I got this error:
```
irb(main):020:0> '/test'.slice!('/test').present?
NoMethodError: undefined method `present?' for "/test":String
	from (irb):20
	from /Users/mark/.rbenv/versions/1.9.3-p547/bin/irb:12:in `<main>'
```

Do you know if Rails is overriding `String` in some way to add `#present?`?

I'm not sure if this is a big issue, but it wasn't clear reading it the first time what was being compared.  Is there a reason why `#start_with?` isn't used here?